### DISCUSSION TITLE
Can I create a custom component with different defaults as the main theme?

### DISCUSSION THREAD

**Discussion by lucksp** [react-native-elements/react-native-elements#3391] [Q&A]
We are working on our setting up default theme, such as where we have a common `Button<ButtonProps>` which is a default, similar to the BASIC BUTTON [in the docs](https://reactnativeelements.com/docs/button).  
We also have a common "secondary" button which would be similar to the OUTLINE BUTTON.  The theme of the design would be nothing like the default, except for font-size/weight.

The font & background colors are inverted, there is a border. None of which exist on the primary/default

Is it possible to make the `FullTheme` include an `OutlineButton`?  Something like:

```
Button: { // this is default Button
      titleStyle: {
        color: colorTheme.backgroundColor,
        fontSize: 16,
        textAlign: "center",
        fontWeight: "700",
        paddingHorizontal: 16,
        paddingVertical: 8,
      },
      buttonStyle: {
        backgroundColor: colorTheme.blue,
        borderColor: colorTheme.blue,
        borderWidth: 0,
        borderRadius: 30,
      },
    },
OutlineButton: { // this is new secondary "default" button which is inverted of default
    titleStyle: {
        color: colorTheme.blue,
        fontSize: 16,
        textAlign: "center",
        fontWeight: "700",
        paddingHorizontal: 16,
        paddingVertical: 8,
      },
      buttonStyle: {
        backgroundColor: 'transparent',
        borderColor: colorTheme.blue,
        borderWidth: 0,
        borderRadius: 30,
      },
}
```

So `<Button />` is the default and `<OutlineButton />` is the secondary?

Or are we stuck with making a custom component for OutlineButton?

```
const OutlineButton = () => {
    const theme = useTheme();

    const style = {
        backgroundColor: 'transparent",
        color: theme.blue,
    };
    return   <Button style={style} />
}
```

thanks for clarification.

**Comment by arpitBhalla** [comment]
This feature is under development. You'd be able to do similar thing with v4

**Reply by lucksp** [reply]
that's awesome!

is the current solution then a custom component?

```
const OutlineButton = () => {
    const theme = useTheme();

    const style = {
        backgroundColor: 'transparent",
        color: theme.blue,
    };
    return   <Button style={style} />
}
```
